### DISCUSSION TITLE
Hello. Where is lq_entry->went_offchip assigned a value?

### DISCUSSION THREAD

**Discussion by szuhzw** [CMU-SAFARI/Hermes#39] [Q&A] [answered]
![image](https://github.com/user-attachments/assets/68582b68-2dd8-40f9-b987-b4e979fff1d3)
@rahulbera Hello. Where is lq_entry->went_offchip assigned a value?

**Comment by rahulbera** [comment, marked as answer]
Hi @szuhzw, LQ entry's `went_offchip` field assignment happens in three cases
- Case 1: when the memory request corresponding to a LQ entry goes off-chip ([cache.cpp:2499](https://github.com/CMU-SAFARI/Hermes/blob/57d00bbb25713379a9340c544f2e8ba6e91e7260/src/cache.cc#L2499))
- Case 2: when the memory request corresponding to a LQ entry gets merged to some other memory request (because they are accessing the same cacheline) at any cache level and the merged request goes off-chip ([cache.cpp:2506](https://github.com/CMU-SAFARI/Hermes/blob/57d00bbb25713379a9340c544f2e8ba6e91e7260/src/cache.cc#L2506))
- Case 3: when the memory request corresponding to a LQ entry gets merged to a memory request from another LQ entry even before accessing the L1D cache, and the merged memory request goes off-chip ([ooo_cpu.cc:2248](https://github.com/CMU-SAFARI/Hermes/blob/57d00bbb25713379a9340c544f2e8ba6e91e7260/src/ooo_cpu.cc#L2248))

Case 2 and 3 (merging cases) can be toggled using the knob `offchip_pred_mark_merged_load`, which is mentioned also in the README.

P.S.: this question could have been resolved by a simple grep in the repo. It's advisable to use this forum for questions/discussions that are not apparently resolved, rather than a tech support.

**Reply by szuhzw** [reply]
Thank you!
